Support full-text index decorator in babel plugin

atdyer opened the pull request:
This closes #5800 by adding support for the `index("full-text")` decorator.

- Check for `index` decorator calls that are passed the string "full-text" and subsequently add the `{ indexed: "full-text" }` property to the schema.
- Update README with description and example of indexing a string field by full text
- Update tests to verify schemas with full-text indexing are properly generated

kneth commented:
@atdyer Thank you for your contribution. It looks good to me, and I have requested the team to review your PR.

I like that you have taken the time to update `README.md` and added tests.

Some test suites will fail since you don't have permission to launch some component - don't worry about that.

Once merged, we will update `CHANGELOG.md` and do a release of the babel plugin.

takameyer commented on packages/babel-plugin/src/index.test.ts:
```diff
@@ -390,6 +390,49 @@ describe("Babel plugin", () => {
+      expect((parsedSchema?.properties.name as PropertySchema).indexed).toBeUndefined();
+    });
+
+    it('ignores `@index()` decorators with invalid parameters', () => {
+      const transformCode = transform({
+        source: `import Realm, { Types, BSON, List, Set, Dictionary, Mixed } from "realm";
+        export class Person extends Realm.Object { @Realm.index("fulltext") name: Realm.Types.String; }`,
+      });
+      const parsedSchema = extractSchema(transformCode);
+
+      expect((parsedSchema?.properties.name as PropertySchema).indexed).toBeUndefined();
+    })
```
I agree that an error should be thrown in this case.

elle-j replied:
Helping the user understand what's wrong and how to correct it also aligns better with the goals of the updated schema parser 👍

takameyer commented on packages/babel-plugin/src/plugin/index.ts:
```diff
@@ -290,6 +290,14 @@ function visitRealmClassProperty(path: NodePath<types.ClassProperty>) {
+  const indexCall =
+    indexDecoratorCall
+      && types.isStringLiteral(indexDecoratorCall.callExpression.arguments[0])
+      && indexDecoratorCall.callExpression.arguments[0].value === 'full-text'
```
It should just forward whatever is given to the schema.  Then at least we get a run-time error.
Unfortunately it's not possible to get type errors here, which would be great.

takameyer reviewed (changes requested):
I would approve this after the comments from @kraenhansen are resolved.

elle-j reviewed (commented):
Thanks for taking the time to understand this issue and propose a solution! 🚀

atdyer commented:
@takameyer I believe I've addressed all your comments - now any string passed into the `@index` decorator as a parameter will be passed through to the final schema. If there's anything I've missed please let me know an I'll fix it up!

kraenhansen reviewed (approved):
As far as I can tell, this just needs an entry in the [changelog](https://github.com/realm/realm-js/blob/main/packages/babel-plugin/CHANGELOG.md).

takameyer reviewed (approved):
Looks good to me 👍🏼

elle-j reviewed (approved):
Nice update!
